pipe_segment/pipeline.py

```diff
@@ -99,67 +95,82 @@ def segmenter_params(self):
-                "description": f"""Created by the pipe-segment:{ver}.
+        )
+
+    @property
+    def segments_output_table(self) -> DateShardedTable:
+        return DateShardedTable(
+            table_id_prefix=self.options.out_segments_table,
+            schema=segment_schema.segment_schema,
+            description=f"""Created by pipe-segment:{__version__}.
                 Daily segments processed in segment step.""",
-            },
-            fragments={
-                "table": self.options.out_fragments_table or self.options.fragments_table,
-                "schema": Fragment.schema,
-                "description": f"""Created by the pipe-segment:{ver}.
+        )
+
+    @property
+    def fragments_output_table(self) -> DateShardedTable:
+        return DateShardedTable(
+            table_id_prefix=self.options.out_fragments_table or self.options.fragments_table,
+            schema=Fragment.schema,
+            description=f"""Created by pipe-segment:{__version__}.
                 Daily fragments processed in segment step.""",
-            })
+        )
 
     @property
-    def write(self):
-        return dict(
-            messages=WriteSink(
-                self.destination_tables["messages"]["table"],
-                self.destination_tables["messages"]["schema"],
-                self.destination_tables["messages"]["description"]
-            ),
-            segments=WriteSink(
-                self.destination_tables["segments"]["table"],
-                self.destination_tables["segments"]["schema"],
-                self.destination_tables["segments"]["description"]
-            ),
-            fragments=WriteSink(
-                self.destination_tables["fragments"]["table"],
-                self.destination_tables["fragments"]["schema"],
-                self.destination_tables["fragments"]["description"]
-            ),
+    def date_sharded_output_tables(self) -> list:
+        return [
+            self.messages_output_table,
+            self.segments_output_table,
+            self.fragments_output_table,
+        ]
+
+    def write_to_date_sharded_table(self, table: DateShardedTable):
+        return WriteSink(
+            sink_table=table.table_id_prefix,
+            schema=table.schema,
+        )
+
+    def prepare_output_tables(self, start_date, end_date):
+        # list_of_days doesn't include the end date. However, in daily mode,
+        # start and end date are the same day.
+        for date in list_of_days(start_date, end_date + dt.timedelta(days=1)):
+            for table in self.date_sharded_output_tables:
+                shard = table.build_shard(date)
+                self.bq_helper.ensure_table_exists(shard)
+                self.bq_helper.run_query(
+                    query=shard.clear_query(),
+                )
+
+        SatelliteOffsetsWrite.prepare_output_tables(
+            self.options,
+            self.cloud_options
         )
```

Comment:
@andres-arana This is failing for VMS because input and output offsets satellite tables are optional and not provided. Everything related to satellite_offsets should be executed only when their tables are provided in the arguments to the pipeline.